src/Test/StateMachine/Z.hs

```diff
@@ -0,0 +1,189 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Test.StateMachine.Z
+-- Copyright   :  (C) 2017, ATS Advanced Telematic Systems GmbH
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Stevan Andjelkovic <stevan@advancedtelematic.com>
+-- Stability   :  provisional
+-- Portability :  portable
+--
+-- This module contains Z-inspried combinators for working with relations. The
```

Comment:
Why not just call it relations? Why mention `Z` at all?
https://en.wikipedia.org/wiki/Relation_(history_of_concept)#Logical_relations_and_database_theory
People have used relations before `Z`...

Reply:
Many operations are taken straight from Z, e.g. overriding and subtraction, which are not part of standard relational terminology.